src/core_modules/capture-ui/DateAndTimeFields/DateField/Date.component.js

```diff
@@ -1,206 +1,76 @@
         this.props.onDateSelectedFromCalendar && this.props.onDateSelectedFromCalendar();
-        document.removeEventListener('click', this.handleDocumentClick);
-    }
-
-    handleDocumentClick({ target }: MouseEvent) {
-        const calendarWrapperDOMElement = this.calendarWrapperDOMElementRef.current;
-
-        if (!calendarWrapperDOMElement) {
-            throw Error('calendar wrapper DOM element not found');
-        }
-
-        if (target === calendarWrapperDOMElement ||
-            (target instanceof Node && calendarWrapperDOMElement.contains(target))) {
-            return;
-        }
-
-        this.hidePopover();
-        document.removeEventListener('click', this.handleDocumentClick);
-    }
-
-    handleTextFieldBlur({ relatedTarget, currentTarget }: SyntheticFocusEvent<HTMLInputElement | HTMLTextAreaElement>) {
-        const calendarWrapperDOMElement = this.calendarWrapperDOMElementRef.current;
-
-        if (!calendarWrapperDOMElement) {
-            throw Error('calendar wrapper DOM element not found');
-        }
-
-        if (relatedTarget === calendarWrapperDOMElement ||
-            (relatedTarget instanceof Node && calendarWrapperDOMElement.contains(relatedTarget))) {
-            document.addEventListener('click', this.handleDocumentClick);
-        } else {
-            this.props.onBlur(currentTarget.value);
-            this.hidePopover();
-        }
-    }
-
-    hidePopover() {
-        this.setState({
-            popoverOpen: false,
-        });
     }
 
     render() {
         const {
             width,
             maxWidth,
             calendarWidth,
-            calendarHeight,
             inputWidth,
             onBlur,
             onFocus,
             onDateSelectedFromCalendar,
+            calendar,
+            label,
+            value,
             ...passOnProps
         } = this.props;
-        const { popoverOpen } = this.state;
+
         const calculatedInputWidth = inputWidth || width;
         const calculatedCalendarWidth = calendarWidth || width;
-        const splittedPassOnProps = DateField.splitPassOnProps(passOnProps);
-        const calculatedCalendarHeight = calendarHeight || 350;
+        const calendarType = calendar || 'gregory';
+        const format = systemSettingsStore.get().dateFormat;
 
         return (
             <div
-                ref={(containerInstance) => { this.containerInstance = containerInstance; }}
                 style={{
                     width,
                     maxWidth,
                 }}
             >
-                { /* // $FlowFixMe */}
-                {/* $FlowFixMe[prop-missing] automated comment */}
-                <DateInput
-                    onFocus={this.handleTextFieldFocus}
-                    onBlur={this.handleTextFieldBlur}
-                    width={calculatedInputWidth}
-                    {...splittedPassOnProps.input}
+                <CalendarInput
+                    {...passOnProps}
```

Comment:
In `passOnProps`, there are a lot of old props that are no longer used and it's quite hard to keep track. Is it possible to clean it up a bit and leave only the props that `CalendarInput` actually uses? You can also clean up unused props from parent components of Date.component.js. Thanks!